lib/storage/search.go

```diff
@@ -427,6 +479,13 @@ func tagFiltersToString(tfs []TagFilter) string {
 func (sq *SearchQuery) Marshal(dst []byte) []byte {
 	dst = encoding.MarshalUint32(dst, sq.AccountID)
 	dst = encoding.MarshalUint32(dst, sq.ProjectID)
+	dst = sq.MarshaWithoutTenant(dst)
+	return dst
+}
+
+// MarshaWithoutTenant appends marshaled sq without AccountID/ProjectID to dst and returns the result.
+// It is expected that TenantToken is already marshaled to dst.
+func (sq *SearchQuery) MarshaWithoutTenant(dst []byte) []byte {
```

Comment:
Looks like this change will break API compatibility. 

 Unmarshal requires Account and Project IDs:

```
	if len(src) < 4 {
		return src, fmt.Errorf("cannot unmarshal AccountID: too short src len: %d; must be at least %d bytes", len(src), 4)
	}
	sq.AccountID = encoding.UnmarshalUint32(src)
	src = src[4:]

	if len(src) < 4 {
		return src, fmt.Errorf("cannot unmarshal ProjectID: too short src len: %d; must be at least %d bytes", len(src), 4)
	}
```